forwarder/vppagent/pkg/vppagent/vppagent.go

```diff
@@ -68,7 +68,8 @@ func (v *VPPAgent) CreateForwarderServer(config *common.ForwarderConfig) forward
 		sdk.Connect(v.endpoint()),
 		sdk.KernelInterfaces(config.NSMBaseDir),
 		sdk.ClearMechanisms(config.NSMBaseDir),
-		sdk.Commit(v.downstreamResync))
+		sdk.Commit(v.downstreamResync),
+		sdk.EthernetContextSetter())
```

Comment:
Are you sure the EthernetContextSetter is after the Commit?

Reply:
>Are you sure the EthernetContextSetter is after the Commit?
 
Yes, It necessary because we want to add arp entry for src interface. If ethernet context was not filled by the user on NSE side, in this case, we need to get mac address of DST itnerface from vpp-agent..
**And we can't get mac address from vpp-agent if the interface was not created.**

How it working:

Case1: User added ethernet context on NSE side remote and local case.

1) For this case `EthernetContextSetter` will do nothing. Because Ethernet context is not empty and ARP entry will be added on converter step (forwarder/vppagent/pkg/converter/kernel_connection_converter.go:189)

Case2: If the ethernet context is empty and it is local case
1) sdk.Commit() will create interfaces
2) sdk.EthernetContextSetter() will get mac address of created DST interface and add arp entry for SRC interface.

Case3: If the ethernet context is empty and it is remote case
On remote side:
1) sdk.Commit() will create DST interface
2) sdk.EthernetContextSetter() will get mac address of created DST interface and put it to EthernetContext

On local side:
1) For this case `EthernetContextSetter` will do nothing. Because Ethernet context is not empty and ARP entry will be added on converter step (forwarder/vppagent/pkg/converter/kernel_connection_converter.go:189)

Reply:
@nickolaev please let me know if you have some questions or ideas for improvements.